src/containers/main/TradeContainer/index.tsx

```diff
@@ -0,0 +1,186 @@
+              <div className={styles.modal_container}>
+                <div className={styles.row}>
+                  <div className={styles.modal_title}>
+                    <h2>Select Token:</h2>
+                  </div>
+                  <div className={styles.modal_title}>
+                    <button onClick={closeTokenModal}>
+                      <h2>X</h2>
+                    </button>
+                  </div>
+                </div>
+                <div>
+                  <TokenButton token="RIC" symbol={iconsCoin.RIC} chooseToken={() => Token('RIC', iconsCoin.RIC)} />
+                  <TokenButton token="DAIx" symbol={iconsCoin.DAI} chooseToken={() => Token('DAIx', iconsCoin.DAI)} />
+                  <TokenButton token="MKRx" symbol={iconsCoin.MKR} chooseToken={() => Token('MKRx', iconsCoin.MKR)} />
+                  <TokenButton token="WBTCx" symbol={iconsCoin.WBTC} chooseToken={() => Token('WBTC', iconsCoin.BTC)} />
+                  <TokenButton token="WETHx" symbol={iconsCoin.ETH} chooseToken={() => Token('WETHx', iconsCoin.ETH)} />
+                  <TokenButton token="MATICx" symbol={iconsCoin.MATIC} chooseToken={() => Token('MATICx', iconsCoin.MATIC)} />
+                  <TokenButton token="SUSHIx" symbol={iconsCoin.SUSHI} chooseToken={() => Token('SUSHIx', iconsCoin.SUSHI)} />
+                  <TokenButton token="IDLEx" symbol={iconsCoin.IDLE} chooseToken={() => Token('IDLEx', iconsCoin.IDLE)} />
+                </div>
+              </div>
+            </div>
+          )
+          :
+          null
+      }
+    </div>
+  );
+};
+
+export const TradeContainer :React.FC = () => {
+  //  const [superTokenFrom] = React.useState();
+  //  const [superTokenTo] = React.useState();
+  const [isTokenA, setIsTokenA] = React.useState(false);
+  const [amountIn, setAmountIn] = React.useState();
+  const [amountOut, setAmountOut] = React.useState();
+  //  const [address] = React.useState();
+  //  const [pool] = React.useState();
+  const [showTokenModal, setTokenModal] = React.useState(false);
+  //  const [showWaringModal] = React.useState(false);
+  const [tokenA, setTokenA] = React.useState({ symbol: undefined, token: 'select token' });
+  const [tokenB, setTokenB] = React.useState({ symbol: undefined, token: 'select token' });
+  //  const [input, setInput] = React.useState('');
+  //  const childRef = React.useRef();
+  
+  const closeTokenModal = () => {
+    setTokenModal(false);
+  };
+
+  const openTokenModal = (choice: boolean) => {
+    setTokenModal(true);
+    setIsTokenA(choice);
+  };
+
+  const updateAmountIn = (e:any) => {
+    const validNumber = new RegExp(/^\d*\.?\d*$/);
+    if (validNumber.test(e.target.value)) {
+      setAmountIn(e.target.value);
+    }
+  };
+
+  const updateAmountOut = (e:any) => {
+    const validNumber = new RegExp(/^\d*\.?\d*$/);
+    if (validNumber.test(e.target.value)) {
+      setAmountOut(e.target.value);
+    }
+  };
+
+  const Swap = () => { 
+  };
```

Comment:
@mindsgn Let me know if you need help with this function, I believe there are some examples in the uniswap sdk docs - https://docs.uniswap.org/sdk/guides/auto-router . It should be straight forward but it took me a while to understand this the first time.

Reply:
@shreyaspapi yes please help me, I'm on the last step which is the swap function.